scripts/inference/cropland_mapping.py

```diff
@@ -41,29 +47,70 @@ def extract(self, image):
     miny = args.miny
     maxx = args.maxx
     maxy = args.maxy
+    epsg = args.epsg
 
     start_date = args.start_date
     end_date = args.end_date
 
-    spatial_extent = BoundingBoxExtent(minx, miny, maxx, maxy)
+    spatial_extent = BoundingBoxExtent(minx, miny, maxx, maxy, epsg)
     temporal_extent = TemporalContext(start_date, end_date)
 
-    backend = BackendContext(Backend.CDSE)
+    backend_context = BackendContext(Backend.FED)
+
+    connection = openeo.connect(
+        "https://openeo.creo.vito.be/openeo/"
+    ).authenticate_oidc()
 
     # Preparing the input cube for the inference
-    input_cube = worldcereal_preprocessed_inputs_gfmap(
-        connection=cdse_connection(),
-        backend_context=backend,
+    inputs = worldcereal_preprocessed_inputs_gfmap(
+        connection=connection,
+        backend_context=backend_context,
         spatial_extent=spatial_extent,
         temporal_extent=temporal_extent,
     )
 
-    # Start the job and download
-    job = input_cube.create_job(
-        title=f"Cropland inference BBOX: {minx} {miny} {maxx} {maxy}",
-        description="Cropland inference using WorldCereal, Presto and GFMAP classifiers",
-        out_format="NetCDF",
+    # Test feature computer
+    presto_parameters = {
+        "rescale_s1": False,  # Will be done in the Presto UDF itself!
+    }
+
+    features = apply_feature_extractor(
+        feature_extractor_class=PrestoFeatureExtractor,
+        cube=inputs,
+        parameters=presto_parameters,
+        size=[
+            {"dimension": "x", "unit": "px", "value": 100},
+            {"dimension": "y", "unit": "px", "value": 100},
+        ],
+        overlap=[
+            {"dimension": "x", "unit": "px", "value": 0},
+            {"dimension": "y", "unit": "px", "value": 0},
+        ],
+    )
+
+    catboost_parameters = {}
+
+    classes = apply_model_inference(
+        model_inference_class=CroplandClassifier,
+        cube=features,
+        parameters=catboost_parameters,
+        size=[
+            {"dimension": "x", "unit": "px", "value": 100},
+            {"dimension": "y", "unit": "px", "value": 100},
+            {"dimension": "t", "value": "P1D"},
+        ],
+        overlap=[
+            {"dimension": "x", "unit": "px", "value": 0},
+            {"dimension": "y", "unit": "px", "value": 0},
+        ],
     )
 
-    job.start_and_wait()
-    job.get_results().download_files(args.output_folder)
+    classes.execute_batch(
+        outputfile=args.output_path,
+        out_format="NetCDF",
+        job_options={
+            "driver-memory": "4g",
+            "executor-memoryOverhead": "12g",
```

Comment:
That's very high. Need to flag this for further profiling. But for now we can leave it like this.

Reply:
From what I have seen it is a potential overkill, 200 km square ran on half the amount of memory for me. @GriffinBabe do you have a job id for a job which needed more memory?